### ISSUE TITLE
Netrender support for gpu rendering 

### ISSUE THREAD

**Issue by zebastian** [buddhi1980/mandelbulber2#529]
The netrender engine should able to handle GPU rendering. 

Things to consider:
- Instead of lines the scheduler needs to schedule tiles which are better handled by gpus 
- all gpu machines in the network would need to be of same type (discrepancy between amd / nvidia)

Maybe implement netrender on a higher level as well for flight and keyframe:
- netrender jobs consists of whole images rendered by each machine. 
--> This would make synchronization easier, as only the finished image has to be streamed into a file on the server. 

this issue has been moved from here: #528

**Comment by buddhi1980** [comment]
I agree with all your points. It is exactly how I'm going to implement it.

**Comment by Toys0125** [comment]
Would it be possible to do both CPU netrending and GPU netrending at the same time if it did the whole images for animations?

**Comment by mclarekin** [comment]
Not sure exactly what you mean but it is possible  but it may not be
feasible:)

Do you mean a network of  CPUs (not using openCL) ,and a network of GPUs
using openCL,  having their workload automatically allocated to each device?

a) if  both the GPU and CPU networks were using openCL,  the rendered
frames will be of the same quality, but, running openCL on CPU's  sometimes
has problems .

b) if  GPUs are single precision openCL and CPU's are using double
precision  c++, then you will need to know how to avoid situations where
you will notice a change in image quality

**Comment by zebastian** [comment]
@Toys0125 probably yes, though you may run into other problems when using heterogenious clients.
If there is some way to implement netrendering code for cpu / gpu combined, there is.

This will get clearer when we implement this in the next release...

**Comment by Michael-Z-Freeman** [comment]
Eh ? So can mandelbulber2 netrender with OpenCL/GPU on macOS ?

**Comment by FacundoMartinezCampos** [comment]
was this fixed? is it possible to use NetRender with GPU?
